lib/shelley/src/Cardano/Wallet/Shelley/Network.hs

```diff
@@ -632,13 +547,16 @@ withNetworkLayerBase tr np conn (versionData, _) action = do
             getRewardMap =
                 fromJustRewards . Map.lookup (Left coin)
 
-    _watchNodeTip nodeTipChan cb = do
-        chan <- dupChan nodeTipChan
+    _watchNodeTip readNodeEraAndTip cb = do
         let toBlockHeader = fromTip getGenesisBlockHash
-        forever $ do
-            header <- toBlockHeader . snd <$> readChan chan
-            bracketTracer (contramap (MsgWatcherUpdate header) tr) $
-                cb header
+        let go oldTip = do
+                tip <- atomically $ do
+                    tip <- readNodeEraAndTip
+                    guard (oldTip /= tip)
+                    return tip
+                cb (toBlockHeader . snd $ tip)
+                go tip
+        go (Just $ AnyCardanoEra ByronEra, TipGenesis )
```

Comment:
we are loosing the `MsgWatcherUpdate` trace from this new approach.

Reply:
👍 thanks